corehq/apps/data_interfaces/tasks.py

```diff
@@ -89,6 +92,13 @@ def run_case_update_rules_for_domain(domain, now=None):
 
         for case_ids in case_id_chunks:
             for case in CaseAccessors(domain).iter_cases(case_ids):
+                time_elapsed = datetime.utcnow() - start_run
+                if time_elapsed.seconds > ONE_DAY_IN_SECONDS:
+                    notify_error(
+                        "Halting rule run for domain %s as it's been running for more than a day." % domain
```

Comment:
yeah so these rules get run every day and while i don't anticipate this scenario to happen often, if it ever does happen it means there's a huge backlog of cases to be closed. so what will happen is it will halt today and then start again tomorrow. knowing the domain should be enough to look into this and see if something is actually wrong or if it's just a large backlog that will be processed eventually.